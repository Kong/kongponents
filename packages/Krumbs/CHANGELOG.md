# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.7.1 (2021-12-17)

**Note:** Version bump only for package @kongponents/krumbs





# 6.7.0 (2021-12-17)


### Features

* **kinput:** add css variable for optional kinput placeholder text color overrides ([#443](https://github.com/Kong/kongponents/issues/443)) ([3b40207](https://github.com/Kong/kongponents/commit/3b40207eabd87b21568ec2697ad39961c2a91b40))





## 6.6.5 (2021-12-17)


### Bug Fixes

* **kpagination:** fix currentPage prop ([#455](https://github.com/Kong/kongponents/issues/455)) ([5e1a801](https://github.com/Kong/kongponents/commit/5e1a801114d2357a2d49dca3bb2571dd055b4119))





## 6.6.4 (2021-12-16)


### Bug Fixes

* **ktextarea:** fix v-model ([#454](https://github.com/Kong/kongponents/issues/454)) ([35f581f](https://github.com/Kong/kongponents/commit/35f581f5c1be13fdebf1a63b8742181de8e19a8b))





## 6.6.3 (2021-12-15)

**Note:** Version bump only for package @kongponents/krumbs





## 6.6.2 (2021-12-10)

**Note:** Version bump only for package @kongponents/krumbs





## 6.6.1 (2021-12-10)


### Bug Fixes

* **kalert:** fix invalid html structure ([#452](https://github.com/Kong/kongponents/issues/452)) ([15eade8](https://github.com/Kong/kongponents/commit/15eade87d0661ae7716a6e32f1fa63279dbf8743))





# 6.6.0 (2021-12-08)


### Features

* **comment:** add global comment in output file ([#451](https://github.com/Kong/kongponents/issues/451)) ([b2f6154](https://github.com/Kong/kongponents/commit/b2f6154c37be30a3d364337f3d87fe40a64fc753))





## 6.5.9 (2021-12-01)


### Bug Fixes

* infinite publish ([d3f275b](https://github.com/Kong/kongponents/commit/d3f275bb668dd95782f6150232cc22a5b95b1bcf))





## [6.5.8](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@6.5.7...@kongponents/krumbs@6.5.8) (2021-12-01)

**Note:** Version bump only for package @kongponents/krumbs





## [6.5.7](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@6.5.6...@kongponents/krumbs@6.5.7) (2021-12-01)

**Note:** Version bump only for package @kongponents/krumbs





## [6.5.6](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@6.5.5...@kongponents/krumbs@6.5.6) (2021-12-01)

**Note:** Version bump only for package @kongponents/krumbs





## [6.5.5](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@6.5.4...@kongponents/krumbs@6.5.5) (2021-12-01)

**Note:** Version bump only for package @kongponents/krumbs





## 6.5.4 (2021-12-01)

**Note:** Version bump only for package @kongponents/krumbs





## 6.5.3 (2021-11-30)

**Note:** Version bump only for package @kongponents/krumbs





## 6.5.2 (2021-11-29)


### Bug Fixes

* **ktable:** allow init client sort ([c3e2a69](https://github.com/Kong/kongponents/commit/c3e2a693974d6aae7f829841af89391c5dd357b2))





## 6.5.1 (2021-11-24)


### Bug Fixes

* **ktable/kcardcatalog:** fetch after init ([#442](https://github.com/Kong/kongponents/issues/442)) ([4a053ae](https://github.com/Kong/kongponents/commit/4a053ae4e39e7df7acc798510c8a2fa9e5ab3f6c))





# 6.5.0 (2021-11-24)


### Features

* **ktable:** add cacheKey prop ([#441](https://github.com/Kong/kongponents/issues/441)) ([743082d](https://github.com/Kong/kongponents/commit/743082df383699b8016d3b6d85497ec77563aaec))





# 6.4.0 (2021-11-24)


### Features

* **ktable:** default sort to asc ([#440](https://github.com/Kong/kongponents/issues/440)) ([95f2e91](https://github.com/Kong/kongponents/commit/95f2e9164c09e50b25f2e334ff30231ab0984503))





# 6.3.0 (2021-11-23)


### Features

* **ktable/kcardcatalog:** make states slottable ([#438](https://github.com/Kong/kongponents/issues/438)) ([43eac93](https://github.com/Kong/kongponents/commit/43eac9337193fe47b28b27576d727e6cbbabad50))





# 6.2.0 (2021-11-23)


### Features

* **ktable/kcardcatalog:** emit cta click events ([#437](https://github.com/Kong/kongponents/issues/437)) ([29eb2a5](https://github.com/Kong/kongponents/commit/29eb2a53cf180935afee1ee10f1b88b462dc29ad))





## 6.1.19 (2021-11-23)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.18 (2021-11-23)


### Bug Fixes

* **ktable:** add missing style ([#435](https://github.com/Kong/kongponents/issues/435)) ([8782dee](https://github.com/Kong/kongponents/commit/8782dee82dafc68c1f956149a249b950359401b1))





## 6.1.17 (2021-11-22)


### Bug Fixes

* fix deps and build process ([#434](https://github.com/Kong/kongponents/issues/434)) ([9fdd180](https://github.com/Kong/kongponents/commit/9fdd180c9816bd4a43d4cd9229fc45b180ba5eed))





## 6.1.16 (2021-11-19)


### Bug Fixes

* **publish:** update node version in publish ([#433](https://github.com/Kong/kongponents/issues/433)) ([26cfe34](https://github.com/Kong/kongponents/commit/26cfe34b6fd9fbb3d61ffdb95412804bac88ba36))





## 6.1.15 (2021-11-19)


### Bug Fixes

* **build:** update lerna ignore flag ([#431](https://github.com/Kong/kongponents/issues/431)) ([768c5e4](https://github.com/Kong/kongponents/commit/768c5e4e78bfca118fe7b2d022778854e076b323))





## 6.1.14 (2021-11-19)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.13 (2021-11-19)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.12 (2021-11-19)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.11 (2021-11-19)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.10 (2021-11-18)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.9 (2021-11-18)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.8 (2021-11-18)


### Bug Fixes

* **css:** box-shadow does not render on ios ([#420](https://github.com/Kong/kongponents/issues/420)) ([c0ba7cb](https://github.com/Kong/kongponents/commit/c0ba7cb90bf3c158a1baa6b879d33230359db830))





## 6.1.7 (2021-11-17)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.6 (2021-11-17)


### Bug Fixes

* **ktextarea:** limit exceeded event ([#418](https://github.com/Kong/kongponents/issues/418)) ([f7cf421](https://github.com/Kong/kongponents/commit/f7cf421a4ded7f5b99eaa5a18d02a27d450020b7))





## 6.1.5 (2021-11-17)

**Note:** Version bump only for package @kongponents/krumbs





## 6.1.4 (2021-11-17)


### Bug Fixes

* general fixes ([#415](https://github.com/Kong/kongponents/issues/415)) ([5b4bb48](https://github.com/Kong/kongponents/commit/5b4bb484d74e75779d39cf901b2cdaa23818e8e7))





## 6.1.3 (2021-11-17)

**Note:** Version bump only for package @kongponents/krumbs





## [6.1.2](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@5.0.8...@kongponents/krumbs@6.1.2) (2021-11-17)

### BREAKING CHANGES
* look and feel changed to match redesign


## [5.0.8](https://github.com/Kong/kongponents/compare/v5.0.7...v5.0.8) (2021-11-15)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.7](https://github.com/Kong/kongponents/compare/v5.0.6...v5.0.7) (2021-10-26)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.6](https://github.com/Kong/kongponents/compare/v5.0.5...v5.0.6) (2021-10-26)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.5](https://github.com/Kong/kongponents/compare/v5.0.4...v5.0.5) (2021-10-13)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.4](https://github.com/Kong/kongponents/compare/v5.0.3...v5.0.4) (2021-10-06)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.3](https://github.com/Kong/kongponents/compare/v5.0.2...v5.0.3) (2021-09-22)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.2](https://github.com/Kong/kongponents/compare/v5.0.1...v5.0.2) (2021-09-15)

**Note:** Version bump only for package @kongponents/krumbs





## [5.0.1](https://github.com/Kong/kongponents/compare/v5.0.0...v5.0.1) (2021-09-13)

**Note:** Version bump only for package @kongponents/krumbs





# [5.0.0](https://github.com/Kong/kongponents/compare/v4.0.2...v5.0.0) (2021-09-13)

**Note:** Version bump only for package @kongponents/krumbs





## [4.0.2](https://github.com/Kong/kongponents/compare/v4.0.1...v4.0.2) (2021-09-09)

**Note:** Version bump only for package @kongponents/krumbs





## [4.0.1](https://github.com/Kong/kongponents/compare/v4.0.0...v4.0.1) (2021-09-07)

**Note:** Version bump only for package @kongponents/krumbs





# 4.0.0 (2021-09-03)


### Bug Fixes

* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))


### Features

* **krumbs:** add maxWidth prop ([#224](https://github.com/Kong/kongponents/issues/224)) ([7dd677d](https://github.com/Kong/kongponents/commit/7dd677d844bd7246fddf261deb3d79b524d24dd0))





# [0.2.0](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.9...@kongponents/krumbs@0.2.0) (2020-06-24)


### Features

* **krumbs:** add maxWidth prop ([#224](https://github.com/Kong/kongponents/issues/224)) ([7dd677d](https://github.com/Kong/kongponents/commit/7dd677d844bd7246fddf261deb3d79b524d24dd0))





## [0.1.9](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.8...@kongponents/krumbs@0.1.9) (2020-03-06)

**Note:** Version bump only for package @kongponents/krumbs





## [0.1.8](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.6...@kongponents/krumbs@0.1.8) (2020-01-16)

**Note:** Version bump only for package @kongponents/krumbs





## [0.1.6](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.5...@kongponents/krumbs@0.1.6) (2019-11-12)

**Note:** Version bump only for package @kongponents/krumbs





## [0.1.5](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.4...@kongponents/krumbs@0.1.5) (2019-11-11)

**Note:** Version bump only for package @kongponents/krumbs





## [0.1.4](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.3...@kongponents/krumbs@0.1.4) (2019-11-08)


### Bug Fixes

* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))





## [0.1.3](https://github.com/Kong/kongponents/compare/@kongponents/krumbs@0.1.2...@kongponents/krumbs@0.1.3) (2019-11-08)

**Note:** Version bump only for package @kongponents/krumbs
