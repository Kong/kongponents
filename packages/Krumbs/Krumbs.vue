<template functional>
  <ul class="krumbs">
    <li
      v-for="item in props.items"
      :key="item.key"
      class="krumb-item"
    >
      <router-link
        v-if="typeof item.to === 'object'"
        :to="item.to"
        :title="item.title"
      >{{ item.text }}</router-link>
      <a
        v-else
        :title="item.title"
        :href="item.to"
        target="_blank">{{ item.text }}</a>
    </li>
  </ul>
</template>

<script>
/**
 * @typedef {Object} Item - breacrumb item holding router-link properties
 * @property {Object|String} to - router-link "to" object or href string
 * @property {string} key - list item key
 * @property {string} text - anchor text
 * @property {string} title - anchor title
 */

export default {
  props: {
    /**
     * @type {{ new(): Item[]}}
     */
    items: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
.krumbs {
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  margin-bottom: 1rem;
  list-style: none;
  border-radius: 0.25rem;
}

.krumbs .krumb-item + .krumb-item::before {
  display: inline-block;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  color: #636c72;
  content: "\203A";
}

.krumbs .krumb-item:last-of-type:after,
.krumbs .krumb span:after {
  display: inline-block;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  color: #636c72;
  content: "\203A";
}
</style>
