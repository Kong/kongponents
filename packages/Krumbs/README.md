```vue
<template>
  <Krumbs :items="routes" />
</template>

<script>
export default {
  data: function () {
    return {
      routes: [
        {
          key: 'root-services',
          to: { name: 'services', params: {}, path: '/services/' },
          title: 'Services',
          text: 'Services'
        },
        {
          key: 'docs',
          to: 'https://docs.konghq.com',
          title: 'Go to Kong Docs',
          text: 'External Link'
        }
      ]
    }
  }
}
</script>
```
