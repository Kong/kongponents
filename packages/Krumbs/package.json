{
  "name": "@kongponents/krumbs",
  "version": "0.1.2",
  "description": "Breadcrumbs component",
  "main": "Krumbs.vue",
  "scripts": {
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "peerDependencies": {
    "vue-router": "^3.0.2"
  },
  "devDependencies": {
    "vue-router": "^3.0.2"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
