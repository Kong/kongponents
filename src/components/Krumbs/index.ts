import { defineComponent, h } from 'vue'
import KBreadcrumbs from '@/components/KBreadcrumbs/KBreadcrumbs.vue'

export default defineComponent({
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Krumbs',
  setup(props, { attrs, slots }) {
    console.warn("The Kongponents 'Krumbs' component is deprecated and will be removed in a future release.\nUpdate all references of 'Krumbs' to 'KBreadcrumbs'.\nKongponent Docs: https://kongponents.konghq.com/components/breadcrumbs.html")

    return () => h(KBreadcrumbs, {
      ...props,
      ...attrs,
    }, slots)
  },
})
