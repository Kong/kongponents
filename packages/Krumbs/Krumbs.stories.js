import Vue from 'vue'
import { storiesOf } from '../../storybook/node_modules/@storybook/vue'
import StoryRouter from '../../storybook/node_modules/storybook-vue-router'

import Krumbs from './Krumbs'

storiesOf('Krumbs', module)
  .addDecorator(StoryRouter())
  .add('Breadcrumbs', () => ({
    components: { Krumbs },
    template: `
      <div><Krumbs :items="routes"></div>
    `,
    data: function () {
      return {
        routes: [
          {
            key: 'root-services',
            to: { name: 'services', params: {}, path: '/services/' },
            title: 'Services',
            text: 'Services'
          },
          {
            key: 'docs',
            to: 'https://docs.konghq.com',
            title: 'Go to Kong Docs',
            text: 'External Link'
          }
        ]
      }
    }
  }))
