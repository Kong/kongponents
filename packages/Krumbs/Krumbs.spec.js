import { mount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Krumbs from '@/Krumbs/Krumbs'

describe('Krumbs', () => {
  it('renders breadcrumb links', () => {
    const localVue = createLocalVue()
    const routes = [{
      path: 'services',
      name: 'root-services',
      redirect: 'services',
      meta: { breadcrumb: 'Services' }
    }]

    const router = new VueRouter({
      mode: 'history',
      routes
    })

    localVue.use(VueRouter)
    const wrapper = mount(Krumbs, {
      localVue,
      context: {
        props: {
          items: [
            {
              key: routes[0].name,
              to: routes[0],
              title: routes[0].meta.breadcrumb,
              text: routes[0].meta.breadcrumb
            },
            {
              key: 'docs',
              to: 'https://docs.konghq.com',
              title: 'Go to Kong Docs',
              text: 'External Link'
            }
          ]
        }
      },
      router
    })

    expect(wrapper.findAll(`ul li`)).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders breadcrumb links without needing a router', () => {
    const wrapper = mount(Krumbs, {
      context: {
        props: {
          items: [
            {
              key: 'docs',
              to: 'https://docs.konghq.com',
              title: 'Go to Kong Docs',
              text: 'External Link'
            }
          ]
        }
      }
    })

    expect(wrapper.findAll(`ul li`)).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
