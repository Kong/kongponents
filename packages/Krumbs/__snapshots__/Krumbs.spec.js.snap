// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Krumbs renders breadcrumb links 1`] = `
<ul class="krumbs">
  <li class="krumb-item"><a href="/services" class="" title="Services">Services</a></li>
  <li class="krumb-item"><a title="Go to Kong Docs" href="https://docs.konghq.com" target="_blank">External Link</a></li>
</ul>
`;

exports[`Krumbs renders breadcrumb links without needing a router 1`] = `
<ul class="krumbs">
  <li class="krumb-item"><a title="Go to Kong Docs" href="https://docs.konghq.com" target="_blank">External Link</a></li>
</ul>
`;
